libs/net/kafka/KafkaListener`.cs

```diff
@@ -158,7 +167,7 @@ public async Task ConsumeAsync(Func<ConsumeResult<TKey, TValue>, Task<ConsumerAc
 
                 _logger.LogInformation("Message received from Kafka topic: '{topic}' key:'{key}'", _currentResult.Topic, _currentResult.Message.Key);
 
-                if (!this.IsPaused)
+                if (!this.IsPaused && (!UseMultiThreads || ReachOrOverLimit))
```

Comment:
When would `!UseMultiThreads != ReachOrOverLimit`?

Reply:
Based on the previous logic they could be different, but it is updated now, thanks